### PR TITLE
[ceres] Added patch to add find_dependency() for suitesparse targets downstream

### LINKED ISSUE #7556: [ceres] cmake project which uses ceres results in build failure
****

**Is your feature request related to a problem? Please describe.**

I am using ceres[core,cxsparse,eigensparse,lapack,suitesparse,tools]:x64-linux, which should be static linked. It is said that I should use it in CMake via
```cmake

    find_package(Ceres CONFIG REQUIRED)
    target_link_libraries(main PRIVATE ceres)

```

However, when I use in my project like

```cmake
cmake_minimum_required(VERSION 3.14)

project(ceres_face_expression)

set(CMAKE_CXX_STANDARD 14)

find_package(pybind11 CONFIG REQUIRED)
#find_package(suitesparse CONFIG REQUIRED)
find_package(Ceres CONFIG REQUIRED)

pybind11_add_module(ceres_exp ceres_exp.cpp ExpCeresFitter.cpp L1SparseCostFunctor.hpp LinearL2CostFunctor.hpp)

target_link_libraries(ceres_exp PRIVATE ceres)
```
There is an error log in CMake:

```
${HOME}/Data/git-task/vcpkg/downloads/tools/cmake-3.14.0-linux/cmake-3.14.0-Linux-x86_64/bin/cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_MAKE_PROGRAM=${HOME}/.linuxbrew/bin/make -DCMAKE_C_COMPILER=${HOME}/.linuxbrew/bin/gcc-9 -DCMAKE_CXX_COMPILER=${HOME}/.linuxbrew/bin/g++-9 -Dpybind11_DIR=${HOME}/Data/anaconda3/envs/nl3dmm/share/cmake/pybind11 -DCMAKE_BUILD_TYPE=Release -DCMAKE_TOOLCHAIN_FILE=${HOME}/Data/git-task/vcpkg/scripts/buildsystems/vcpkg.cmake -G "CodeBlocks - Unix Makefiles" /tmp/tmp.sNqrB0xSG9
-- Found installed version of Eigen: ${HOME}/Data/git-task/vcpkg/installed/x64-linux/share/eigen3
-- Found required Ceres dependency: Eigen version 3.3.7 in ${HOME}/Data/git-task/vcpkg/installed/x64-linux/include/eigen3
-- Found Ceres version: 1.14.0 installed in: ${HOME}/Data/git-task/vcpkg/installed/x64-linux with components: [EigenSparse, SparseLinearAlgebraLibrary, SuiteSparse, CXSparse, SchurSpecializations, OpenMP, Multithreading]
-- Configuring done
CMake Error at ${HOME}/Data/git-task/vcpkg/scripts/buildsystems/vcpkg.cmake:198 (_add_library):
  Target "ceres_exp" links to target "SuiteSparse::suitesparseconfig" but the
  target was not found.  Perhaps a find_package() call is missing for an
  IMPORTED target, or an ALIAS target is missing?
Call Stack (most recent call first):
  ${HOME}/Data/anaconda3/envs/nl3dmm/share/cmake/pybind11/pybind11Tools.cmake:131 (add_library)
  CMakeLists.txt:11 (pybind11_add_module)


CMake Error at ${HOME}/Data/git-task/vcpkg/scripts/buildsystems/vcpkg.cmake:198 (_add_library):
  Target "ceres_exp" links to target "SuiteSparse::amd" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
Call Stack (most recent call first):
  ${HOME}/Data/anaconda3/envs/nl3dmm/share/cmake/pybind11/pybind11Tools.cmake:131 (add_library)
  CMakeLists.txt:11 (pybind11_add_module)


CMake Error at ${HOME}/Data/git-task/vcpkg/scripts/buildsystems/vcpkg.cmake:198 (_add_library):
  Target "ceres_exp" links to target "SuiteSparse::btf" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
Call Stack (most recent call first):
  ${HOME}/Data/anaconda3/envs/nl3dmm/share/cmake/pybind11/pybind11Tools.cmake:131 (add_library)
  CMakeLists.txt:11 (pybind11_add_module)


CMake Error at ${HOME}/Data/git-task/vcpkg/scripts/buildsystems/vcpkg.cmake:198 (_add_library):
  Target "ceres_exp" links to target "SuiteSparse::camd" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
Call Stack (most recent call first):
  ${HOME}/Data/anaconda3/envs/nl3dmm/share/cmake/pybind11/pybind11Tools.cmake:131 (add_library)
  CMakeLists.txt:11 (pybind11_add_module)


CMake Error at ${HOME}/Data/git-task/vcpkg/scripts/buildsystems/vcpkg.cmake:198 (_add_library):
  Target "ceres_exp" links to target "SuiteSparse::ccolamd" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
Call Stack (most recent call first):
  ${HOME}/Data/anaconda3/envs/nl3dmm/share/cmake/pybind11/pybind11Tools.cmake:131 (add_library)
  CMakeLists.txt:11 (pybind11_add_module)


CMake Error at ${HOME}/Data/git-task/vcpkg/scripts/buildsystems/vcpkg.cmake:198 (_add_library):
  Target "ceres_exp" links to target "SuiteSparse::colamd" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
Call Stack (most recent call first):
  ${HOME}/Data/anaconda3/envs/nl3dmm/share/cmake/pybind11/pybind11Tools.cmake:131 (add_library)
  CMakeLists.txt:11 (pybind11_add_module)


CMake Error at ${HOME}/Data/git-task/vcpkg/scripts/buildsystems/vcpkg.cmake:198 (_add_library):
  Target "ceres_exp" links to target "SuiteSparse::cholmod" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
Call Stack (most recent call first):
  ${HOME}/Data/anaconda3/envs/nl3dmm/share/cmake/pybind11/pybind11Tools.cmake:131 (add_library)
  CMakeLists.txt:11 (pybind11_add_module)


CMake Error at ${HOME}/Data/git-task/vcpkg/scripts/buildsystems/vcpkg.cmake:198 (_add_library):
  Target "ceres_exp" links to target "SuiteSparse::cxsparse" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
Call Stack (most recent call first):
  ${HOME}/Data/anaconda3/envs/nl3dmm/share/cmake/pybind11/pybind11Tools.cmake:131 (add_library)
  CMakeLists.txt:11 (pybind11_add_module)


CMake Error at ${HOME}/Data/git-task/vcpkg/scripts/buildsystems/vcpkg.cmake:198 (_add_library):
  Target "ceres_exp" links to target "SuiteSparse::klu" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
Call Stack (most recent call first):
  ${HOME}/Data/anaconda3/envs/nl3dmm/share/cmake/pybind11/pybind11Tools.cmake:131 (add_library)
  CMakeLists.txt:11 (pybind11_add_module)


CMake Error at ${HOME}/Data/git-task/vcpkg/scripts/buildsystems/vcpkg.cmake:198 (_add_library):
  Target "ceres_exp" links to target "SuiteSparse::ldl" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
Call Stack (most recent call first):
  ${HOME}/Data/anaconda3/envs/nl3dmm/share/cmake/pybind11/pybind11Tools.cmake:131 (add_library)
  CMakeLists.txt:11 (pybind11_add_module)


CMake Error at ${HOME}/Data/git-task/vcpkg/scripts/buildsystems/vcpkg.cmake:198 (_add_library):
  Target "ceres_exp" links to target "SuiteSparse::umfpack" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
Call Stack (most recent call first):
  ${HOME}/Data/anaconda3/envs/nl3dmm/share/cmake/pybind11/pybind11Tools.cmake:131 (add_library)
  CMakeLists.txt:11 (pybind11_add_module)


CMake Error at ${HOME}/Data/git-task/vcpkg/scripts/buildsystems/vcpkg.cmake:198 (_add_library):
  Target "ceres_exp" links to target "SuiteSparse::spqr" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
Call Stack (most recent call first):
  ${HOME}/Data/anaconda3/envs/nl3dmm/share/cmake/pybind11/pybind11Tools.cmake:131 (add_library)
  CMakeLists.txt:11 (pybind11_add_module)


CMake Error at ${HOME}/Data/git-task/vcpkg/scripts/buildsystems/vcpkg.cmake:198 (_add_library):
  Target "ceres_exp" links to target "SuiteSparse::suitesparseconfig" but the
  target was not found.  Perhaps a find_package() call is missing for an
  IMPORTED target, or an ALIAS target is missing?
Call Stack (most recent call first):
  ${HOME}/Data/anaconda3/envs/nl3dmm/share/cmake/pybind11/pybind11Tools.cmake:131 (add_library)
  CMakeLists.txt:11 (pybind11_add_module)


CMake Error at ${HOME}/Data/git-task/vcpkg/scripts/buildsystems/vcpkg.cmake:198 (_add_library):
  Target "ceres_exp" links to target "SuiteSparse::amd" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
Call Stack (most recent call first):
  ${HOME}/Data/anaconda3/envs/nl3dmm/share/cmake/pybind11/pybind11Tools.cmake:131 (add_library)
  CMakeLists.txt:11 (pybind11_add_module)


CMake Error at ${HOME}/Data/git-task/vcpkg/scripts/buildsystems/vcpkg.cmake:198 (_add_library):
  Target "ceres_exp" links to target "SuiteSparse::btf" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
Call Stack (most recent call first):
  ${HOME}/Data/anaconda3/envs/nl3dmm/share/cmake/pybind11/pybind11Tools.cmake:131 (add_library)
  CMakeLists.txt:11 (pybind11_add_module)


CMake Error at ${HOME}/Data/git-task/vcpkg/scripts/buildsystems/vcpkg.cmake:198 (_add_library):
  Target "ceres_exp" links to target "SuiteSparse::camd" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
Call Stack (most recent call first):
  ${HOME}/Data/anaconda3/envs/nl3dmm/share/cmake/pybind11/pybind11Tools.cmake:131 (add_library)
  CMakeLists.txt:11 (pybind11_add_module)


CMake Error at ${HOME}/Data/git-task/vcpkg/scripts/buildsystems/vcpkg.cmake:198 (_add_library):
  Target "ceres_exp" links to target "SuiteSparse::ccolamd" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
Call Stack (most recent call first):
  ${HOME}/Data/anaconda3/envs/nl3dmm/share/cmake/pybind11/pybind11Tools.cmake:131 (add_library)
  CMakeLists.txt:11 (pybind11_add_module)


CMake Error at ${HOME}/Data/git-task/vcpkg/scripts/buildsystems/vcpkg.cmake:198 (_add_library):
  Target "ceres_exp" links to target "SuiteSparse::colamd" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
Call Stack (most recent call first):
  ${HOME}/Data/anaconda3/envs/nl3dmm/share/cmake/pybind11/pybind11Tools.cmake:131 (add_library)
  CMakeLists.txt:11 (pybind11_add_module)


CMake Error at ${HOME}/Data/git-task/vcpkg/scripts/buildsystems/vcpkg.cmake:198 (_add_library):
  Target "ceres_exp" links to target "SuiteSparse::cholmod" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
Call Stack (most recent call first):
  ${HOME}/Data/anaconda3/envs/nl3dmm/share/cmake/pybind11/pybind11Tools.cmake:131 (add_library)
  CMakeLists.txt:11 (pybind11_add_module)


CMake Error at ${HOME}/Data/git-task/vcpkg/scripts/buildsystems/vcpkg.cmake:198 (_add_library):
  Target "ceres_exp" links to target "SuiteSparse::cxsparse" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
Call Stack (most recent call first):
  ${HOME}/Data/anaconda3/envs/nl3dmm/share/cmake/pybind11/pybind11Tools.cmake:131 (add_library)
  CMakeLists.txt:11 (pybind11_add_module)


CMake Error at ${HOME}/Data/git-task/vcpkg/scripts/buildsystems/vcpkg.cmake:198 (_add_library):
  Target "ceres_exp" links to target "SuiteSparse::klu" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
Call Stack (most recent call first):
  ${HOME}/Data/anaconda3/envs/nl3dmm/share/cmake/pybind11/pybind11Tools.cmake:131 (add_library)
  CMakeLists.txt:11 (pybind11_add_module)


CMake Error at ${HOME}/Data/git-task/vcpkg/scripts/buildsystems/vcpkg.cmake:198 (_add_library):
  Target "ceres_exp" links to target "SuiteSparse::ldl" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
Call Stack (most recent call first):
  ${HOME}/Data/anaconda3/envs/nl3dmm/share/cmake/pybind11/pybind11Tools.cmake:131 (add_library)
  CMakeLists.txt:11 (pybind11_add_module)


CMake Error at ${HOME}/Data/git-task/vcpkg/scripts/buildsystems/vcpkg.cmake:198 (_add_library):
  Target "ceres_exp" links to target "SuiteSparse::umfpack" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
Call Stack (most recent call first):
  ${HOME}/Data/anaconda3/envs/nl3dmm/share/cmake/pybind11/pybind11Tools.cmake:131 (add_library)
  CMakeLists.txt:11 (pybind11_add_module)


CMake Error at ${HOME}/Data/git-task/vcpkg/scripts/buildsystems/vcpkg.cmake:198 (_add_library):
  Target "ceres_exp" links to target "SuiteSparse::spqr" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
Call Stack (most recent call first):
  ${HOME}/Data/anaconda3/envs/nl3dmm/share/cmake/pybind11/pybind11Tools.cmake:131 (add_library)
  CMakeLists.txt:11 (pybind11_add_module)


-- Generating done
-- Build files have been written to: /tmp/tmp.sNqrB0xSG9/cmake-build-tryvcpkg
```
So I modified my CMakeLists.txt as
``` cmake
cmake_minimum_required(VERSION 3.14)

project(ceres_face_expression)

set(CMAKE_CXX_STANDARD 14)

find_package(pybind11 CONFIG REQUIRED)
find_package(suitesparse CONFIG REQUIRED)
find_package(Ceres CONFIG REQUIRED)

pybind11_add_module(ceres_exp ceres_exp.cpp ExpCeresFitter.cpp L1SparseCostFunctor.hpp LinearL2CostFunctor.hpp)

target_link_libraries(ceres_exp PRIVATE ceres SuiteSparse::amd SuiteSparse::btf SuiteSparse::klu SuiteSparse::ldl)
```
Then the output log is
```
-- Using VCPKG FindBLAS. Remove if CMake has been updated to account for Threads in OpenBLAS!
-- A library with LAPACK API not found. Please specify library location.
-- Found installed version of Eigen: ${HOME}/Data/git-task/vcpkg/installed/x64-linux/share/eigen3
-- Found required Ceres dependency: Eigen version 3.3.7 in ${HOME}/Data/git-task/vcpkg/installed/x64-linux/include/eigen3
-- Found Ceres version: 1.14.0 installed in: ${HOME}/Data/git-task/vcpkg/installed/x64-linux with components: [EigenSparse, SparseLinearAlgebraLibrary, SuiteSparse, CXSparse, SchurSpecializations, OpenMP, Multithreading]
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/tmp.sNqrB0xSG9/cmake-build-tryvcpkg
```

Please note that it didn't find LAPACK API.

Now the project compiles and builds, when when I execute it as
```
Python 3.5.2 (default, Nov 12 2018, 13:43:14) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import ceres_exp
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: /tmp/tmp.sNqrB0xSG9/cmake-build-tryvcpkg/ceres_exp.cpython-35m-x86_64-linux-gnu.so: undefined symbol: dpotrf_
>>> 
```
So it doesn't find LAPACK API.

**Proposed solution**

- What is the proper way to use the static linked x64-linux ceres? I want to know what should be written into CMakeLists.txt.

- Is there anything going wrong with vcpkg's FindLAPACK and FindBLAS? I think x64-linux tends to static link external libraries and dynamic link c runtime, so I expect my project to use a static version of lapack and doesn't need external .so file when running, but vcpkg cannot find the lapack and openblas libraries installed by vcpkg.

**Describe alternatives you've considered**

I know that the same CMakeLists.txt works with linuxbrew, with CMAKE_PREFIX_PATH set to linuxbrew's installation path.

Thanks!

### PR DESCRIPTION
**Pull request by venabled**
Should address / close microsoft/vcpkg#7556

The vcpkg port of `suitesparse` uses the cmake-ified version, which is nice in that it actually declares modern cmake targets for the individual libraries. Only issue is that libraries that build on suitesparse may not be aware of this, and not declare / find these targets in their own cmake. 

For example, the `sfm` feature in `opencv4` uses `ceres`, and fails to link correctly due to the inability to find the `suitesparse` targets. 

This pull requests adds a `find_dependency` block in `CeresConfig.cmake.in` so that users of ceres will be able to find the locations of the suitesparse libs/targets in downstream cmake projects, if `suitesparse` is a selected feature of `ceres`

- What does your PR fix?
   - Fixes issue #7556

- Which triplets are supported/not supported? Have you updated the CI baseline?
   - Tested on `x64-osx` and `x64-linux`

- Does your PR follow the [maintainer guide]
   - To the best of my ability to parse it.

### CODE REVIEW

File: ports/ceres/portfile.cmake [hunk 1]
```diff
@@ -10,6 +10,10 @@ if(VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Linux")
     set(ADDITIONAL_PATCH "0004_blas_linux_fix.patch")
 endif()
 
+if("suitesparse" IN_LIST FEATURES)
```
Comment (JackBoosY) [inline ports/ceres/portfile.cmake:13]: I think we should use `vcpkg_check_features` to judge whether to enable this feature, and pass it as a parameter to `CMakeLists.txt`, and finally in _CeresConfig.cmake.in_ according to the variable through `configure_file` to make the judgment condition is **TRUE** or **FALSE**.

**Review by JackBoosY — CHANGES_REQUESTED** [review]
Please update the verion info. See [documentation](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#update-the-version-field-in-the-control-file-of-any-modified-ports).

File: ports/ceres/portfile.cmake (hunk 1 above)
Comment (JackBoosY) [reply ports/ceres/portfile.cmake:13]: See [example](https://github.com/microsoft/vcpkg/blob/master/ports/czmq/fix-dependencies.patch).

Comment (venabled) [reply ports/ceres/portfile.cmake]: I bumped the version, used `vcpkg_check_features`, and modified patch `0003` to use the `@` macro/substitution for detecting if `SUITESPARSE` is set.

File: ports/ceres/portfile.cmake [hunk 2]
```diff
@@ -61,11 +45,6 @@ vcpkg_configure_cmake(
         -DEXPORT_BUILD_DIR=ON
```
Comment (JackBoosY) [inline ports/ceres/portfile.cmake:45]: Missing add `${FEATURE_OPTIONS}` here.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/ceres/portfile.cmake (hunk 2 above)
Comment (venabled) [reply ports/ceres/portfile.cmake:46]: Sorry, I don't quite understand how this built with my last commit, without this line in there. Updated to include this line.

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
Already successfully tested all features in `x86-windows`, `x64-windows`, `x64-windows-static`.
LGTM, thanks for this PR!

**Comment by strega-nil** [comment]
Alright, awesome! Thanks @venabled :)

**Pull request merged**